app/models/validations/sales/sale_information_validations.rb

```diff
@@ -23,4 +23,17 @@ def validate_previous_property_unit_type(record)
       record.errors.add :fromprop, I18n.t("validations.sale_information.previous_property_type.property_type_bedsit")
     end
   end
+
+  def validate_discounted_ownership_value(record)
+    return unless record.mortgage && record.value && record.deposit && record.ownershipsch && (record.discount || record.grant)
+
```

Comment:
don't we also want this validation to run in the case that there is neither a discount nor a grant (i.e. `rent_to_buy_full_ownership?` == true) or is that covered by a separate validation?

Reply:
Have a look now?